clients/src/main/java/org/apache/kafka/clients/consumer/internals/SubscriptionState.java

```diff
@@ -270,12 +262,12 @@ public boolean partitionsAutoAssigned() {
         return !this.subscription.isEmpty();
     }
 
-    public void consumed(TopicPartition tp, long offset) {
-        assignedState(tp).consumed(offset);
+    public void position(TopicPartition tp, long offset) {
```

Comment:
I agree this is the main question with this simplification. However, keep in mind that we cannot send more than one fetch request at a time because we don't know the following offset to fetch from. So to get any advantage from allowing the fetched position to get farther than a single fetch ahead of the consumed position, we would need to initiate fetches after the last fetch returned from the server, but before the results were returned to the user. I don't see a lot of opportunity for optimization here, but I could be missing something. I think instead the way for consumers to tune the amount of data fetched is with `fetch.min.bytes`.